app/src/main/java/com/iic/shopingo/ui/OnboardingActivity.java

```diff
@@ -0,0 +1,110 @@
+package com.iic.shopingo.ui;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.ActionBarActivity;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import butterknife.InjectView;
+import com.facebook.Session;
+import com.facebook.SessionState;
+import com.facebook.UiLifecycleHelper;
+import com.facebook.widget.LoginButton;
+import com.iic.shopingo.R;
+
+public class OnboardingActivity extends ActionBarActivity {
+
+  private static final String LOG_TAG = OnboardingActivity.class.getSimpleName();
+
+  @InjectView(R.id.onboarding_login_btn)
+  LoginButton facebookLoginBtn;
+
+  private UiLifecycleHelper facebookLifecycleHelper;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_onboarding);
+
+    facebookLifecycleHelper = new UiLifecycleHelper(this, new Session.StatusCallback() {
```

Comment:
Yeah but that would mean that the activity would have a "call" method which is not clear as to what its purpose is, I'd rather have the onSessionStateChanged method in the activity.